Onboarding enablers changes

cZikos opened the pull request:
This is an update to documentation for onboarding services due to recent changes.

JirkaAichler commented on docs/extend/extend-apiml/api-mediation-onboard-an-existing-java-rest-api-service-without-spring-boot-with-zowe-api-mediation-layer.md:
````diff
@@ -256,6 +256,10 @@ After you add API Mediation Layer integration endpoints, you are ready to add se
     ```yaml
     serviceId: hellospring
     title: HelloWorld Spring REST API
+    eureka:
````
Remove this part. This wasn't added by me and we would need to test it.

cZikos replied:
Removed

JirkaAichler commented on docs/extend/extend-apiml/api-mediation-onboard-an-existing-java-rest-api-service-without-spring-boot-with-zowe-api-mediation-layer.md:
```diff
@@ -266,13 +270,11 @@ After you add API Mediation Layer integration endpoints, you are ready to add se
     routedServices:
     - gatewayUrl: api/v1
       serviceUrl: /hellospring/api/v1
-    
```
In this case we need api-doc endpoint to access swagger. Return back this part.

cZikos replied:
Done

JirkaAichler commented on docs/extend/extend-apiml/api-mediation-onboard-an-existing-java-rest-api-service-without-spring-boot-with-zowe-api-mediation-layer.md:
```diff
@@ -312,7 +314,20 @@ After you add API Mediation Layer integration endpoints, you are ready to add se
 
         **Tip:** We recommend that you provide a specific default value of the `title`.
         Use a title that describes the service instance so that the end user knows the specific purpose of the service instance.
+        
```
Remove this part. I don't know if it is needed or not.

cZikos replied:
I am not sure which part you refer to. Title, Description, etc. are different than apiInfo.title, apiInfo.description that were replaced.

BrandonJenkins14 commented:
FYI, looks like there are 3 broken links in this branch that will need to be fixed before you can merge to master. Let me know if you need help interpreting the link report. Thanks!

janan07 commented on docs/extend/extend-apiml/api-mediation-onboard-an-existing-java-rest-api-service-without-spring-boot-with-zowe-api-mediation-layer.md:
```diff
@@ -381,18 +382,31 @@ After you add API Mediation Layer integration endpoints, you are ready to add se
+        
+    * **apiInfo.documentationUrl**
+
+        (Optional) Link to external documentation, if needed. Can coexist with Swagger documentation.
```
Explain 'coexist'. Can this be re-written? 
Example:
"the link to the external documentation can be included along with the Swagger documentation"

cZikos replied:
I replaced it

janan07 commented on docs/extend/extend-apiml/api-mediation-onboard-a-spring-boot-rest-api-service.md:
```diff
@@ -388,7 +389,21 @@ As an API service developer, you set multiple configuration settings in your app
+          (Optional) Link to external documentation, if needed. Can coexist with Swagger documentation
+      * `eureka.instance.metadata-map.apiml.apiInfo.swaggerUrl`
+      
+          (Optional) Specifies the HTTP or HTTPS address where the Swagger JSON document that provides the API documentation for this API is available.                   
```
This is unclear. Is it clear enough to simply say,
"(Optional) Specifies the HTTP or HTTPS address where the Swagger JSON document is available. "

cZikos replied:
This was a copy from _docs/extend/extend-apiml/api-mediation-onboard-an-existing-rest-api-service-without-code-changes.md. 
I simplified it everywhere in the code.

janan07 commented on docs/extend/extend-apiml/api-mediation-onboard-a-spring-boot-rest-api-service.md:
```diff
@@ -388,7 +389,21 @@ As an API service developer, you set multiple configuration settings in your app
+          
+      * `eureka.instance.metadata-map.apiml.apiInfo.documentationUrl`
+      
+          (Optional) Link to external documentation, if needed. Can coexist with Swagger documentation
```
Clarify what 'coexisit' means here.

janan07 commented on docs/extend/extend-apiml/api-mediation-onboard-an-existing-java-rest-api-service-without-spring-boot-with-zowe-api-mediation-layer.md:
```diff
@@ -381,18 +382,31 @@ After you add API Mediation Layer integration endpoints, you are ready to add se
+
+    * **apiInfo.swaggerUrl**
+
+        (Optional) Specifies the HTTP or HTTPS address where the Swagger JSON document
```
Lines 403-404 are unclear. Please clarify.

JirkaAichler commented on docs/extend/extend-apiml/api-mediation-onboard-a-spring-boot-rest-api-service.md:
```diff
@@ -388,7 +389,21 @@ As an API service developer, you set multiple configuration settings in your app
+      * `eureka.instance.metadata-map.apiml.apiInfo.apiId`
+      
+          Specifies the API identifier that is registered in the API Mediation Layer installation. The API ID uniquely identifies the API in the API Mediation Layer.
+          The same API can be provided by multiple service. The API ID can be used to locate the same APIs that are provided by different services. The creator of the API defines this ID.
```
Please make sure that the parameter description for all enablers is consistent. ' The same API can be provided by multiple service.' should be changed to ' The same API can be provided by multiple serviceS.' There is probably more occurrences.

cZikos replied:
Fixed the "s" for all Docs.

cZikos commented:
> I fixed 2/4 links. I am having trouble diagnosing 2 of the links, and it will be easier to diagnose once this is merged to master. Feel free to merge.
> 
> FYI @cZikos, I found out that the link you corrected from "gizafoundation", to "zowe.org" was actually a legacy link that is still correct. I reverted that back to https://projectgiza.org/Downloads/post_download.html I don't blame you for trying to fix it though!

Thanks Brandon, I will work on the rest of the links. Yes, my bad, I thought that this was the  link the report was referring to.

JirkaAichler commented on docs/user-guide/api-mediation/api-mediation-api-catalog.md:
```diff
@@ -6,7 +6,7 @@ As an application developer, use the API Catalog to view what services are runni
 * Ensure that the service you would like to view in the API Catalog is configured to be displayed in the API Catalog.
 
-**Note:** For more information about how to configure the API Catalog in the application.yml, see: [Add API Onboarding Configuration](topics/api-mediation/api-mediation-onboard-an-existing-spring-boot-rest-api-service.md).
+**Note:** For more information about how to configure the API Catalog in the application.yml, see: [Add API Onboarding Configuration](topics/api-mediation/api-mediation-onboard-an-existing-sprint-boot-rest-api-service.md).
```
Should it be 'spring'?

janan07 replied:
Fixed.

cZikos commented:
This PR is by mistake to be merged to master. PR #418 was created instead, to be merged to docs-staging.
